src/functionalTest/java/uk/gov/hmcts/reform/userprofileapi/AddRolesToExistingUserFuncTest.java

```diff
@@ -204,4 +206,60 @@ public void rdcc_418_ac2_should_update_user_status_from_suspended_to_active() th
         log.info("user updated to:" + actual1.getIdamStatus());
     }
 
+    @Test
+    public void should_throw_412_while_add_roles_with_invalid_roles_passed() throws Exception {
+        UserProfileCreationData data = createUserProfileData();
+        List<String> roles = new ArrayList<>();
+        roles.add(puiUserManager);
+        String email = idamClient.createUser(roles);
+        data.setEmail(email);
+        UserProfileCreationResponse userProfileCreationResponse = createUserProfile(data, HttpStatus.CREATED);
+
+        RoleName rolesToBeAdded = new RoleName("pui-org-manager");
+        Set<RoleName> rolesToAdd = new HashSet<>();
+        rolesToAdd.add(rolesToBeAdded);
+
+        UpdateUserProfileData updateUserProfileData = new UpdateUserProfileData();
+        updateUserProfileData.setRolesAdd(rolesToAdd);
+
+        UserProfileRolesResponse deleteResourceResp =
+                testRequestHandler.sendDelete(
+                        updateUserProfileData,
+                        HttpStatus.OK,
+                        requestUri + "/" + userProfileCreationResponse.getIdamId(),
+                        UserProfileRolesResponse.class);
+        assertThat(deleteResourceResp).isNotNull();
+        RoleAdditionResponse roleAdditionResponse = deleteResourceResp.getRoleAdditionResponse();
+        assertThat(roleAdditionResponse.getIdamMessage()).isEqualTo("One or more of the roles provided does not exist.");
+        assertThat(roleAdditionResponse.getIdamStatusCode()).isEqualTo("412");
+    }
+
+    @Test
+    public void should_throw_412_while_add_roles_with_already_unassigned_roles_passed() throws Exception {
+        UserProfileCreationData data = createUserProfileData();
+        List<String> roles = new ArrayList<>();

```

Comment:
already assigned roles?
could you please reuse the code in both test cases most of the code is common apart from role name

Reply:
done